homeassistant/components/publibike/config_flow.py

```diff
@@ -0,0 +1,83 @@
+"""Config flow for the PubliBike Public API integration."""
+
+from __future__ import annotations
+
+import logging
+
+from pypublibike.publibike import PubliBike
+from requests import RequestException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_validation as cv
+
+from .const import (
+    BATTERY_LIMIT,
+    BATTERY_LIMIT_DEFAULT,
+    DOMAIN,
+    LATITUDE,
+    LONGITUDE,
+    STATION_ID,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Optional(BATTERY_LIMIT, default=BATTERY_LIMIT_DEFAULT): vol.All(
+            cv.positive_int, vol.Range(1, 100)
+        ),
```

Comment:
I must admit, I _am_ a bit confused about how this should be set up properly. I want all of these options to be available during initial configuration - when I use OptionsFlow for this one option though, it would get moved to the post-creation configuration, right? Moreover, I am not entirely sure how to make it explicit that _either_ coordinates _or_ station id should be provided (as in, one of the two is required). Any hints would be appreciated - I couldn't really find a relevant example to compare this against  :cry:

Reply:
OK, I finally got this working - battery and location moved to options flow. Tests updated accordingly.